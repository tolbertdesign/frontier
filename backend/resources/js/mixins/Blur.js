export default {
  methods: {
    blur () {
      document.getElementById('app').style.filter = 'blur(4px)'
    },
    unBlur () {
      document.getElementById('app').style.filter = 'none'
    },
  },
}
