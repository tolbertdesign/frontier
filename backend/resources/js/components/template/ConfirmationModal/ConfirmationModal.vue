<template>
  <div class="bg-grey-lightest pb-2 mx-auto shadow-md rounded-xl max-w-xs">
    <div
      class="modal-card"
      style="width: auto;"
    >
      <button
        type="button"
        class="modal-close inner-modal-close modal-close-btn-position is-large absolute bg-transparent mb-4"
        aria-label="close"
        @click="cancel"
      >&times;</button>

      <p class="text-center text-lg leading-normal mt-18 max-w-xs">
        {{ lang.about_to_delete }}
      </p>
      <p class="text-center font-semibold text-lg leading-normal mt-4 mb-6">
        {{ lang.continue }}
      </p>

      <div
        class="controls px-6"
      >
        <div class="text-center mt-5 pb-1">
          <div class="max-w-md mx-0 md:mx-auto">
            <div>
              <button
                type="submit"
                class="button is-danger is-rounded inline-block mb-4 w-64 shadow"
                @click="confirmed"
              >
                {{ lang.delete }}
              </button>
              <br>
              <button
                type="button"
                class="button border-grey-dark text-grey-dark is-rounded inline-block mb-4 w-64 is-outlined shadow"
                @click="cancel"
              >
                {{ lang.cancel }}
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
import Blur from '@/mixins/Blur'
export default {
  name: 'ConfirmationModal',
  status: 'prototype',
  release: '1.0.0',
  mixins: [Blur],
  computed: {
    lang () {
      return this.$store.state.lang
    },
  },
  methods: {
    cancel () {
      this.$emit('close')
      this.unBlur()
      return false
    },
    confirmed () {
      this.$emit('close')
      this.unBlur()
      this.$emit('confirmed')
    },
  },
}
</script>
