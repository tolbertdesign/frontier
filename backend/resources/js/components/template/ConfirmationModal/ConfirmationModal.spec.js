import { shallowMount, createLocalVue } from '@vue/test-utils'
import ConfirmationModal from './ConfirmationModal.vue'
import Blur from '@/mixins/Blur'

const localVue = createLocalVue()

const $store = {
  state: {
    lang: {
      about_to_delete: 'about to delete',
      continue: 'continue',
      delete: 'delete',
      cancel: 'cancel',
    },
  },
}

describe('ConfirmationModal', () => {
  it('Snapshop test for ConfirmationModal', () => {
    const wrapper = shallowMount(ConfirmationModal, {
      localVue,
      mixins: {
        Blur,
      },
      mocks: {
        $store,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
