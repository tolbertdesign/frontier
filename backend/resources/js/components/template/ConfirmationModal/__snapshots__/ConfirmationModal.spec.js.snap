// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationModal Snapshop test for ConfirmationModal 1`] = `
<div class="bg-grey-lightest pb-2 mx-auto shadow-md rounded-xl max-w-xs">
  <div class="modal-card" style="width: auto;"><button type="button" aria-label="close" class="modal-close inner-modal-close modal-close-btn-position is-large absolute bg-transparent mb-4">Ã—</button>
    <p class="text-center text-lg leading-normal mt-18 max-w-xs">about to delete</p>
    <p class="text-center font-semibold text-lg leading-normal mt-4 mb-6">continue</p>
    <div class="controls px-6">
      <div class="text-center mt-5 pb-1">
        <div class="max-w-md mx-0 md:mx-auto">
          <div><button type="submit" class="button is-danger is-rounded inline-block mb-4 w-64 shadow">
              delete
            </button> <br> <button type="button" class="button border-grey-dark text-grey-dark is-rounded inline-block mb-4 w-64 is-outlined shadow">
              cancel
            </button></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
